src/fugw/utils.py

```diff
@@ -189,6 +190,68 @@ def _add_dict(d, new_d):
     return d
 
 
+def init_plan_dense(
+    n_source,
+    n_target,
+    weights_source=None,
+    weights_target=None,
+    method="entropic",
+):
+    """Initialize transport plan with dense tensor.
+
+    Generate a matrix satisfying the constraints of a transport plan.
+    In particular, marginal constraints on lines and columns are satisfied.
+
+    Parameters
+    ----------
+    n_source: int
+        Number of source points
+    n_target: int
+        Number of target points
+    weights_source: torch.Tensor of size(n_source), optional, defaults to None
+        Source weights used in entropic init
+    weights_target: torch.Tensor of size(n_target), optional, defaults to None
+        Target weights used in entropic init
+    method: str, optional, defaults to "entropic"
+        Method to use for initialization.
+        Can be "entropic", "permutation" or "identity".
+        If "entropic", weights_source and weights_target must be provided ;
+        the initial plan is then given by the product of the two arrays.
+        If "permutation", the initial plan is the solution to a 1D
+        optimal transport problem between two random arrays, which can be
+        understood as a soft permutation between source and target points.
+        If "identity", the number of source and target points must be equal ;
+        the initial plan is then the identity matrix.
+
+    Returns
+    -------
+    init_plan: torch.Tensor of size(n_source, n_target)
+    """
+
+    if method == "identity":
+        assert n_source == n_target, (
+            "Number of source and target points must be equal "
+            "when using identity initialization."
+        )
+        plan = torch.eye(n_source, dtype=torch.float32)
```

Comment:
Maybe you should replace ``dtype=torch.float32`` with ``dtype=torch.float64`` to avoid casting issues.

Reply:
The default type we use in `_make_tensor()` is `torch.float32` (or `torch.int32`), which is why I chose this type here :slightly_smiling_face: 

https://github.com/alexisthual/fugw/blob/ffb6fc0b378def7dd0d3bf989b084c4c2513efd7/src/fugw/utils.py#L38-L55

Reply:
Okay let's keep ``float32`` for coherence then ! 👍